docs: C/C++ quickstart results in "No rule to make target 'quickstart'. Stop."

### What happened?

It appears that the target name `quickstart` is wrong in [C/C++ quickstart](https://arrow.apache.org/adbc/main/cpp/quickstart.html#quickstart),

```console
$ mamba install cmake compilers libadbc-driver-manager libadbc-driver-sqlite
$ mkdir build
$ cd build
$ cmake ../docs/source/cpp/recipe
$ cmake --build . --target quickstart
gmake: *** No rule to make target 'quickstart'.  Stop.
```

It appears that it should be `recipe-quickstart`.

https://github.com/apache/arrow-adbc/blob/317c9c90c690a8dedb2f4557337c12ad9cd673df/docs/source/cpp/recipe/CMakeLists.txt#L38

```console
$ cmake --build . --target recipe-quickstart
[ 14%] Building C object _deps/nanoarrow-build/CMakeFiles/nanoarrow.dir/src/nanoarrow/array.c.o
[ 28%] Building C object _deps/nanoarrow-build/CMakeFiles/nanoarrow.dir/src/nanoarrow/schema.c.o
[ 42%] Building C object _deps/nanoarrow-build/CMakeFiles/nanoarrow.dir/src/nanoarrow/array_stream.c.o
[ 57%] Building C object _deps/nanoarrow-build/CMakeFiles/nanoarrow.dir/src/nanoarrow/utils.c.o
[ 71%] Linking C static library libnanoarrow.a
[ 71%] Built target nanoarrow
[ 85%] Building CXX object CMakeFiles/recipe-quickstart.dir/quickstart.cc.o
[100%] Linking CXX executable recipe-quickstart
[100%] Built target recipe-quickstart
$ ./recipe-quickstart
Got -1 rows
Result schema: struct<THEANSWER: int64>
Got a batch with 1 rows
THEANSWER[0] = 42
```

It had changed from `quickstart` since the following fix.
 
* https://github.com/apache/arrow-adbc/commit/2daf2263bb1aad36f19267699ce9d1abf97b4ed4#diff-705bee359657852c868ed28459dab14e0142e46ce69f2dbb0a2b30552e25f1cc

### Stack Trace

Not related.

### How can we reproduce the bug?

https://arrow.apache.org/adbc/main/cpp/quickstart.html#quickstart

```bash
mamba install cmake compilers libadbc-driver-manager libadbc-driver-sqlite
mkdir build
cd build
cmake ../docs/source/cpp/recipe
cmake --build . --target quickstart
```

### Environment/Setup

* Ubuntu 22.04.5 LTS
* mamba 2.1.1 (Miniforge)

--- a/docs/source/cpp/recipe/quickstart.cc
+++ b/docs/source/cpp/recipe/quickstart.cc
@@ -46,8 +46,8 @@
 ///    mkdir build
 ///    cd build
 ///    cmake ../docs/source/cpp/recipe
-///    cmake --build . --target quickstart
-///    ./quickstart
+///    cmake --build . --target recipe-quickstart
+///    ./recipe-quickstart
 ///
 /// .. _CMake: https://cmake.org/
 

